test/Main.purs

```diff
@@ -30,10 +31,27 @@ spec = do
       testElement (H.div "" "Foo") $
         text >>= shouldEqual "Foo"
 
-  describe "find" $
+  describe "find" do
     it "gets an element wrapper by a selector" $
       testComponent def $
         find ".bar" >> text >>= shouldEqual "Bar"
+    it "crashes when multiple elements are found" $
+      testComponent def $
+        try (find ".qux p") >>= case _ of
+          Left err -> message err `shouldEqual` "Expected a single element matching '.qux p', but found 3"
+          Right _ -> fail "Expected find to crash"
+    it "crashes when zero elements are found" $
+      testComponent def $
+        try (find ".qux a") >>= case _ of
+          Left err -> message err `shouldEqual` "Expected a single element matching '.qux a', but found 0"
+          Right _ -> fail "Expected find to crash"
+
+  describe "findAll" $
+    it "gets multiple elements" $
+      testComponent def do
+        findAll ".qux" >> length >>= shouldEqual 2
+        findAll ".qux" >> findAll "p" >> length >>= shouldEqual 3
```

Comment:
Interesting that this just works

Reply:
What in particular?

Reply:
In enzyme, if `wrapper.find('.qux')` returns many elements, then you can do `quz.find('p')` and it returns all the paragraph elements under all the `.qux`elements. It's like a monadic operation. Just surprising that the JS library works like that.

And then that makes me wonder, is `forEach` needed? Doesn't `findAll x >> f` do the same thing?

Reply:
The fact that `w.find(".qux").find("p")` works by "aggregating" the results of second `find` and ends up equivalent to `w.find(".qux p")` is just CSS semantics. And if you think about it, there is no other way it could work.

But it's not like a monadic operation (I'm assuming you're talking about the list monad). `ElementWrapper` doesn't work like that _in general_, it's only the `find` function that is special (and maybe some other similar ones).

And for example, `text` cannot work like that, because it needs to return a string. And sure, in JavaScript, it _could_ return either a string or an array of strings, but that would be soooo tedious to use! So instead it crashes. A more or less valid choice I think.

And yes, `forEach` is definitely needed. For the very reason that some functions don't work for multiple elements. For example, `find("button") >> forEach (simulate "click")` works, but `find("button") >> simulate "click"` doesn't.

Reply:
Ah, that makes sense 👍🏻